src/Servers/Kestrel/Core/test/HttpParserTests.cs

```diff
@@ -394,6 +394,22 @@ public void ParseRequestLineSplitBufferWithoutNewLineDoesNotUpdateConsumed()
             Assert.Equal(buffer.End, examined);
         }
 
+        [Fact]
+        public void ParseRequestLineTlsOverHttp()
+        {
+            var parser = CreateParser(_nullTrace);
+            var buffer = ReadOnlySequenceFactory.CreateSegments(new byte[] { 0x16, 0x03, 0x01, 0x02, 0x00, 0x01, 0x00, 0xfc, 0x03, 0x03, 0x03, 0xca, 0xe0, 0xfd, 0x0a });
+
+            var requestHandler = new RequestHandler();
+
+            var badHttpRequestException = Assert.Throws<BadHttpRequestException>(() =>
+            {
+                parser.ParseRequestLine(requestHandler, buffer, out var consumed, out var examined);
+            });
+
+            Assert.Equal(badHttpRequestException.StatusCode, StatusCodes.Status400BadRequest);
```

Comment:
This test doesn't distinguish between the new behavior and the old. You need to check the message or enum to ensure it's the new behavior, correct?

Reply:
Yes, you are right. Is it OK for you guys to check the message to make sure the outcome is correct? I usually try to avoid string comparison to distinguish between errors.

Reply:
I added a comparison for the reason - this is much better than string comparison.

Reply:
Checking the enum is adequate.